test/stablecoin_converter.js

```diff
@@ -0,0 +1,109 @@
+      const orderResult = (await stablecoinConverter.orders.call(user_1,id))
+      assert.equal((orderResult.sellAmount).toNumber(), 20, "sellAmount was stored incorrectly")
+      assert.equal((orderResult.buyAmount).toNumber(), 10, "buyAmount was stored incorrectly")
+      assert.equal((orderResult.sellToken).toNumber(), 1, "sellToken was stored incorrectly")
+      assert.equal((orderResult.buyToken).toNumber(), 0, "buyToken was stored incorrectly")
+      assert.equal(orderResult.sellOrderFlag, true, "sellTokenFlag was stored incorrectly")
+      assert.equal((orderResult.validFrom).toNumber(), currentStateIndex.toNumber() + 1, "validFrom was stored incorrectly")
+      assert.equal((orderResult.validUntil).toNumber(), 3, "validUntil was stored incorrectly")
+    })
+  })
+  describe("cancelOrder", () => {
+    it("places orders, then cancels it and orders status", async () => {
+      const stablecoinConverter = await StablecoinConverter.new(2**16-1)
+
+      const id = await stablecoinConverter.placeOrder.call(0, 1, true, 3, 10, 20, {from: user_1})
+      await stablecoinConverter.placeOrder(0, 1, true, 3, 10, 20, {from: user_1})
+      const currentStateIndex = await stablecoinConverter.getCurrentStateIndex()
+      await stablecoinConverter.cancelOrder(id, {from: user_1})
+
+      assert.equal(((await stablecoinConverter.orders.call(user_1,id)).validUntil).toNumber(), currentStateIndex, "validUntil was stored incorrectly")
+    })
+  })
+  describe("freeStorageOfOrder", () => {
+    it("places a order, then cancels and deletes it", async () => {
+      const stablecoinConverter = await StablecoinConverter.new(2**16-1)
+
+      const id = await sendTxAndGetReturnValue(stablecoinConverter.placeOrder, 0, 1, true, 3, 10, 20)
+      await stablecoinConverter.cancelOrder(id)
+      await waitForNSeconds(300)
+      await stablecoinConverter.freeStorageOfOrder(id)
+
+      assert.equal((await stablecoinConverter.orders(user_1, id)).sellAmount, 0, "sellAmount was stored incorrectly")
+    })
+    it("fails to delete non-canceled order", async () => {
+      const stablecoinConverter = await StablecoinConverter.new(2**16-1)
+      const currentStateIndex = await stablecoinConverter.getCurrentStateIndex()
+
+      const id = await sendTxAndGetReturnValue(stablecoinConverter.placeOrder, 0, 1, true, currentStateIndex + 3, 10, 20)
+      await truffleAssert.reverts(stablecoinConverter.freeStorageOfOrder(id), "Order is still valid")
+    })
+    it("fails to delete canceled order in same stateIndex", async () => {
+      const stablecoinConverter = await StablecoinConverter.new(2**16-1)
+    
+      const id = await sendTxAndGetReturnValue(stablecoinConverter.placeOrder, 0, 1, true, 3, 10, 20)
+      await stablecoinConverter.cancelOrder(id)
+      await truffleAssert.reverts(stablecoinConverter.freeStorageOfOrder(id), "Order is still valid")
+    })
+  })
+  describe("addToken()", () => {
+    it("Anyone can add tokens", async () => {
```

Comment:
Do we really want to allow anyone to add tokens?

Reply:
Nope, we have to come up with a specific solution later. I made a note in the open question section here:
https://docs.google.com/document/d/1UuMjran_ajk5wntm8pMjkI4C3A1Z-7hevSUa-4Uej2U/edit?ts=5d49a26e